### PR TITLE
configurable backend with env variables

### PR DESCRIPTION
**Pull request by Falence**
This makes the backend configurable by setting up environment variables to avoid the hardcoding of values such as ports, keys e.t.c

### CODE REVIEW

**Review by ivange94 — CHANGES_REQUESTED** [review]
The way I'd like this implemented is that we have a `config` package that hides all the logic of reading configurations. Then a config object like

```
type Config struct {
    Host string
    Port int
}

var DefaultConfig = getConfig()

func getConfig() {
    config := Config{}
    config.Host = os.Getenv("PORT")
    config.Port = os.Getenv("Host")

    return &config
}
```
Then inside main you can do 
```
port := config.DefaultConfig.Port
router.Logger.Fatal(router.Start(port))
```

**Comment by Falence** [comment]
Oh I see. This actually makes a lot of sense. Thank you.
I'll work on it

**Comment by Falence** [comment]
I've modified the PR. Kindly let me know what you think.

File: README.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ Website for DevConfCM (devconf.cm)
 #### Run
 `cd devconfcm` 
 
+Create a `.env` to house all environment variables (`PORT` e.t.c)
```
Comment (ivange94) [inline README.md]: Document all environment variables needed.

File: pkg/config/config.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"os"
+
+	_ "github.com/joho/godotenv/autoload"
+)
+
+type Config struct {
+	Host string
+	Port string
```
Comment (ivange94) [inline pkg/config/config.go]: Port should be an integer

File: pkg/config/config.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+func getConfigs() Config { 
+	return Config{
+		Port: ":" + os.Getenv("PORT"),
```
Comment (ivange94) [inline pkg/config/config.go]: Just `Port: os.Getenv("PORT")`

**Review by ivange94 — CHANGES_REQUESTED** [review]
Looks good. Just minor changes and a recommendation to use viper for this. With viper you can provide a config.json with default environment variables. These variables can be overridden by environment variables or command line flags. `config.json` will be tracked by git and should never hold sensitive data like usernames and passwords. We'll always use env variables for those. See a quick entry to viper https://dev.to/techschoolguru/load-config-from-file-environment-variables-in-golang-with-viper-2j2d

**Comment by Falence** [comment]
> Looks good. Just minor changes and a recommendation to use viper for this. With viper you can provide a config.json with default environment variables. These variables can be overridden by environment variables or command line flags. `config.json` will be tracked by git and should never hold sensitive data like usernames and passwords. We'll always use env variables for those. See a quick entry to viper https://dev.to/techschoolguru/load-config-from-file-environment-variables-in-golang-with-viper-2j2d

I've modified the PR.
Thank you for this. I must say, I learnt a lot already.

File: app.env [hunk 4]
```diff
@@ -0,0 +1,3 @@
+PORT=5000
+HOST=
+TEST_DB=test.db
```
Comment (ivange94) [inline app.env]: Add new line

File: app.env [hunk 5]
```diff
@@ -0,0 +1,3 @@
+PORT=5000
+HOST=
```
Comment (ivange94) [inline app.env]: If the default will be an empty string then you can just leave it out.

File: README.md [hunk 6]
```diff
@@ -15,8 +15,11 @@ Website for DevConfCM (devconf.cm)
 #### Run
 `cd devconfcm` 
 
+The `app.env` file houses all non-sensitive development variables. For now we have: `PORT`, `HOST` and `TEST_DB`.
+For sensitive variables such as `USERNAME`, `PASSWORD` e.t.c, put them in your system's evironment variables.
```
Comment (ivange94) [inline README.md:25]: Add a section for variables like this

```
### Configuration variables

* PORT: Port application will be listening to
* HOST: 
* DB_URL: Url for db
```
We don't need the `TEST_DB` config.

File: pkg/config/config.go [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+func getConfigs() Config {
+	configs, err := LoadConfigs(".") // path: root folder
+	if err != nil {
+		log.Fatal("Unable to log configs: ", err.Error())
+	}
+	return configs
+}
+
+var DefaultConfigs = getConfigs()
```
Comment (ivange94) [inline pkg/config/config.go]: This should be singular `DefaultConfig`

File: pkg/router/router.go [hunk 8]
```diff
@@ -32,5 +35,6 @@ func start() {
 	router.File("/", "website/build/index.html")
 
 	// start server
-	router.Logger.Fatal(router.Start(":5000"))
+	port := ":" + strconv.Itoa(config.DefaultConfigs.Port)
```
Comment (ivange94) [inline pkg/router/router.go]: This looks cleaner

```
port := config.DefaultConfigs.Port
url := fmt.Sprintf(":%v", port)
```

**Review by ivange94 — CHANGES_REQUESTED** [review]
Just minor comments.

**Comment by Falence** [comment]
Resolved!

**Comment by ivange94** [comment]
Thank you @Falence

**Comment by Falence** [comment]
@ivange94 I'm looking forward to more tasks and more code reviews

**Pull request merged**